tests/lldp/test_lldp_syncd.py

```diff
@@ -250,7 +250,56 @@ def test_lldp_entry_table_content(
         # Add assertions to compare specific fields between LLDP_ENTRY_TABLE and lldpctl output
 
 
-# Test case 3: Verify LLDP_ENTRY_TABLE after interface flap
+# Test case 3: Verify LLDP_ENTRY_TABLE after restart syncd and orchagent
+@pytest.mark.disable_loganalyzer
+def test_lldp_entry_table_after_syncd_orchagent(
+    duthosts, enum_rand_one_per_hwsku_frontend_hostname, db_instance
+):
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    if duthost.facts['asic_type'] == "vs":
+        pytest.skip("Skip this test case for virtual testbed")
+    # Verify LLDP_ENTRY_TABLE keys match show lldp table output at the start of test
+    keys_match = wait_until(30, 5, 0, check_lldp_table_keys, duthost, db_instance)
+    if not keys_match:
+        assert keys_match, "LLDP_ENTRY_TABLE keys do not match 'show lldp table' output"
+
+    logging.info("Stop and start syncd and swss on DUT")
+    duthost.shell("docker restart syncd")
+    duthost.shell("docker restart swss")
+    wait_until(150, 5, 60, duthost.critical_services_fully_started)
+    # Wait until all interfaces are up and lldp entries are populated
+    lldp_entry_keys = get_lldp_entry_keys(db_instance)
+    for interface in lldp_entry_keys:
+        result = wait_until(120, 2, 0, verify_lldp_entry, db_instance, interface)
+        entry_content = get_lldp_entry_content(db_instance, interface)
+        pytest_assert(
+            result,
+            "After restart swss and syncd, interface {} LLDP_ENTRY_TABLE entry is not correct:{}".format(
+                interface, entry_content
+            ),
+        )
+    # To get lldp entry keys again after all interfaces are up
+    lldp_entry_keys = get_lldp_entry_keys(db_instance)
+    lldpctl_output = get_lldpctl_output(duthost)
+    show_lldp_table_int_list = get_show_lldp_table_output(duthost)
+    lldpctl_interfaces = lldpctl_output["lldp"]["interface"]
+    assert_lldp_interfaces(
+        lldp_entry_keys, show_lldp_table_int_list, lldpctl_interfaces
+    )
+    for interface in get_lldp_entry_keys(db_instance):
+        entry_content = get_lldp_entry_content(db_instance, interface)
```

Comment:
This seems repeating code in this file, could we move them into a function? Too much code and logic inside this test to make it hard to read through, and distracting from the main function.